### PR TITLE
feat(interaction): Add new inputType.touch.preventDefault option

### PR DESCRIPTION
**Pull request by netil**
## Issue

#82

## Details

- Control option to prevent option during touch events.
- Added event bidings test codes
- Making clear the condition on attaching events

### CODE REVIEW

**Comment by netil** [comment]
Changed interface as :

```js
interaction: {
   enabled: true,
   inputType: {
       touch: false

       // or declare preventDefault explicitly.
       // In this case touch inputType is enabled by default
       touch: {
           preventDefault: false

           // or threshold pixel value (pixel moved from touchstart to touchmove)
           preventDefault: 5
       }
   }
}
```

File: src/interactions/interaction.js [hunk 1]
```diff
@@ -127,27 +108,55 @@ extend(ChartInternal.prototype, {
...
+					} else if (preventThreshold !== null) {
+						startPx = currentXY;
+					}
+				} else if (eventType === "touchmove") {
+					if (isPrevented || startPx === true || (
+						preventThreshold !== null && Math.abs(startPx - currentXY) >= preventThreshold
+					)) {
+						// once prevented, keep prevented during whole 'touchmove' context
+						startPx = true;
+						event.preventDefault();
```
Comment (sculove) [inline src/interactions/interaction.js]: The preventDefault method is called each time the touchmove event is called.

Comment (sculove) [inline src/interactions/interaction.js]: How about using dirty checking?

Comment (sculove) [inline src/interactions/interaction.js]: I was confused.  preventDefault is fired every time.
LGTM

**Pull request merged**